authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OAuth1RestSourceAuthorizationService.kt

```diff
@@ -0,0 +1,183 @@
+        val req = createRequest("DELETE", authConfig.deregistrationEndpoint!!, RestOauth1AccessToken(user.accessToken!!, user.refreshToken), user.sourceType)
+
+        return httpClient.newCall(req).execute().use { response ->
+            when (response.code) {
+                200, 204 -> true
+                400, 401, 403 -> false
+                else -> throw HttpBadGatewayException("Cannot connect to ${authConfig.deregistrationEndpoint}: HTTP status ${response.code}")
+            }
+        }
+    }
+
+    override fun deRegisterUser(user: RestSourceUser): RestSourceUser {
+        revokeToken(user)
+        val userDTO = userMapper.fromEntity(user)
+        userDTO.isAuthorized = false
+        return userRepository.update(user, userDTO)
+    }
+
+    override fun getAuthorizationEndpointWithParams(sourceType: String, callBackUrl: String): String {
+        logger.info("Getting auth endpoint..")
+        val authConfig = configMap[sourceType]
+                ?: throw HttpBadRequestException("client-config-not-found", "Cannot find client configurations for source-type $sourceType")
+
+        val tokens = this.requestToken(authConfig.preAuthorizationEndpoint, RestOauth1AccessToken(""), sourceType)
+        val params = mapOf<String, String?>(
+            OAUTH_ACCESS_TOKEN to tokens?.token,
+            OAUTH_ACCESS_TOKEN_SECRET to tokens?.tokenSecret,
+            OAUTH_CALLBACK to callBackUrl
+        )
+
+        return Url(authConfig.authorizationEndpoint, params).getUrl()
+    }
+
+    fun requestToken(tokenEndpoint: String?, tokens: RestOauth1AccessToken, sourceType: String): RestOauth1AccessToken? {
+        val req = createRequest("POST", tokenEndpoint.orEmpty(), tokens, sourceType)
+
+        return httpClient.newCall(req).execute().use { response ->
+            when (response.code) {
+                200 -> response.body?.string()
+                        ?.let { tokenReader.readValue<RestOauth1AccessToken>(parseParams(it)) }
+                        ?: throw HttpBadGatewayException("Service did not provide a result")
+                400, 401, 403 ->{
+                    logger.info(response.body?.string())
+                    null }
+                else -> throw HttpBadGatewayException("Cannot connect to ${tokenEndpoint}: HTTP status ${response.code}")
+            }
+        }
+    }
+
+    fun createRequest(method: String, url: String, tokens: RestOauth1AccessToken, sourceType: String): Request {
+        val authConfig = configMap[sourceType]
+                ?: throw HttpBadRequestException("client-config-not-found", "Cannot find client configurations for source-type ${sourceType}")
+        var params = this.getAuthParams(authConfig, tokens.token, tokens.tokenVerifier)
+        val signature = OauthSignature(url, params, method, authConfig.clientSecret, tokens.tokenSecret).getEncodedSignature()
+        params[OAUTH_SIGNATURE] = signature
```

Comment:
This can just directly be assigned 
```suggestion
        params[OAUTH_SIGNATURE] = OauthSignature(url, params, method, authConfig.clientSecret, tokens.tokenSecret)
            .getEncodedSignature()
```